olive/workflows/run/__main__.py

```diff
@@ -3,21 +3,39 @@
 # Licensed under the MIT License.
 # --------------------------------------------------------------------------
 import argparse
+import sys
+from pathlib import Path
 
 from olive.common.utils import set_tempdir
 from olive.workflows import run
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser("Olive Workflow: Custom Run")
-    parser.add_argument("--config", type=str, help="Path to json config file", required=True)
+    parser.add_argument("--olive-config", type=str, help="Path to Olive (json) config file", required=False)
+    parser.add_argument("--run-config", "--config", type=str, help="Path to json config file", required=True)
     parser.add_argument("--setup", help="Whether run environment setup", action="store_true")
-    parser.add_argument("--data_root", help="The data root path for optimization", required=False)
+    parser.add_argument("--data-root", "--data_root", help="The data root path for optimization", required=False)
     parser.add_argument("--tempdir", type=str, help="Root directory for tempfile directories and files", required=False)
 
     args = parser.parse_args()
 
+    if "--config" in sys.argv:
+        print(  # noqa:T201
+            "WARNING: Option '--config' is deprecated and will be removed in future versions. "
+            "Use '--run-config' instead."
+        )
+    if "--data_root" in sys.argv:
```

Comment:
is there a reason for preferring `-` over `_`?

Reply:
I read about the convention to use '-' in the parameter which python converts it to '_' - justification being that it distinguishes the use cases and avoids accidental changes with operations like find/replace.

Unfortunately, I can't seem to find the actual source link again. :(

For the moment, both conventions are supported.